### PR TITLE
[chore] Apply spotless

### PR DESCRIPTION
**Pull request by andreaTP**
As we are going for GA and, after the big changes for sync/async the GH history is already a bit compromised, I think this is the right time to introduce this change.

How to review:

- the spotless configuration is at this commit: https://github.com/microsoft/kiota-java/commit/c095aefa465a0e1ecfbb6a97af35449c35345867
- feel free to restart from there and add any different configuration you might like and `./gradlew spotlessApply` to see how it looks

### CODE REVIEW

File: components/abstractions/src/main/java/com/microsoft/kiota/BaseRequestBuilder.java [hunk 1]
```diff
@@ -1,47 +1,55 @@
...
 import java.util.Objects;
 
-import jakarta.annotation.Nonnull;
-
 /** Base class for request builders */
 public abstract class BaseRequestBuilder {
     /** Path parameters for the request */
-    @Nonnull
-    protected HashMap<String, Object> pathParameters;
+    @Nonnull protected HashMap<String, Object> pathParameters;
```
Comment (baywet) [inline components/abstractions/src/main/java/com/microsoft/kiota/BaseRequestBuilder.java:10]: is there a way to configure the behaviour? I'm not sure this kind of change is super helpful

File: components/abstractions/src/main/java/com/microsoft/kiota/RequestAdapter.java [hunk 2]
```diff
@@ -32,9 +32,12 @@ public interface RequestAdapter {
      * @param <ModelType> the type of the response model to deserialize the response into.
      * @return the deserialized response model.
      */
-    @Nullable
-    @SuppressWarnings("LambdaLast")
-    <ModelType extends Parsable> ModelType send(@Nonnull final RequestInformation requestInfo, @Nonnull final ParsableFactory<ModelType> factory, @Nullable final HashMap<String, ParsableFactory<? extends Parsable>> errorMappings);
+    @Nullable @SuppressWarnings("LambdaLast")
```
Comment (baywet) [inline components/abstractions/src/main/java/com/microsoft/kiota/RequestAdapter.java:35]: same thing here, I'm not a big fan of that change either. But I might be biased by dotnet.

**Review by baywet — COMMENTED** [review]
Thanks for setting this up. It's a great improvement to our dev loop.
I did have a couple of questions.
Also, do you think it'd be possible to enforce things are properly formatted as part of the build/CI process?

File: components/abstractions/src/main/java/com/microsoft/kiota/RequestAdapter.java (hunk 2 above)
Comment (andreaTP) [reply components/abstractions/src/main/java/com/microsoft/kiota/RequestAdapter.java:35]: No strong opinions here, I let you take the lead, I'm happy with whatever format as long as it's automated.

**Comment by andreaTP** [comment]
> Also, do you think it'd be possible to enforce things are properly formatted as part of the build/CI process?

This is already done, `./gradlew build` will fail if files are not formatted.

File: build/tmp/spotless-register-dependencies [hunk 3]
```diff
@@ -0,0 +1 @@
+1
```
Comment (baywet) [inline build/tmp/spotless-register-dependencies:1]: should this file be here? or should we add it to the git ignore?

File: components/abstractions/src/main/java/com/microsoft/kiota/BaseRequestBuilder.java (hunk 1 above)
Comment (baywet) [reply components/abstractions/src/main/java/com/microsoft/kiota/BaseRequestBuilder.java:10]: Actually, after [reading the documentation](https://github.com/diffplug/spotless/tree/main/plugin-gradle#formatannotations) I retract my objection here. This annotation qualifies the Type of the field, this is why it's inline. An annotation qualifying the field itself would be at the line above. It's different from dotnet, but the rationale makes sense. I'll just have to get used to it. Thanks for sharing the context.

File: components/abstractions/src/main/java/com/microsoft/kiota/RequestAdapter.java (hunk 2 above)
Comment (baywet) [reply components/abstractions/src/main/java/com/microsoft/kiota/RequestAdapter.java]: (fixes the configuration for SuppressWarnings, resolving)

File: build/tmp/spotless-register-dependencies (hunk 3 above)
Comment (baywet) [reply build/tmp/spotless-register-dependencies]: after some local testing, it appears this file is generated on build and can be removed safely.

**Review by baywet — APPROVED** [review]

**Comment by andreaTP** [comment]
Amazing! Thanks a ton for making this happen @baywet !!! ❤️

**Pull request merged**